test/integration/kubernetes_deploy_test.rb

```diff
@@ -106,13 +106,19 @@ def test_dead_pods_in_old_replicaset_are_ignored
       deployment["spec"]["template"]["spec"]["activeDeadlineSeconds"] = 1
     end
 
-    sleep 1 # make sure to hit DeadlineExceeded on at least one pod
+    initial_failed_pod_count = 0
+    while initial_failed_pod_count < 1
+      pods = kubeclient.get_pods(namespace: @namespace, label_selector: "name=web,app=hello-cloud")
+      initial_failed_pod_count = pods.count { |pod| pod.status.phase == "Failed" }
+    end
 
     deploy_fixtures("hello-cloud", subset: ["web.yml.erb", "configmap-data.yml"])
     pods = kubeclient.get_pods(namespace: @namespace, label_selector: "name=web,app=hello-cloud")
-    running_pods, not_running_pods = pods.partition { |pod| pod.status.phase == "Running" }
-    assert_equal 1, running_pods.size
-    assert not_running_pods.size >= 1
+    running_pod_count = pods.count { |pod| pod.status.phase == "Running" }
+    final_failed_pod_count = pods.count { |pod| pod.status.phase == "Failed" }
+
+    assert_equal 1, running_pod_count
+    assert final_failed_pod_count >= initial_failed_pod_count # failed pods not cleaned up
```

Comment:
Next time you can use `assert_operator 5, :<=, 4` that generates a slightly better failure message